web/impact/impact/tests/test_graphql.py

```diff
@@ -582,10 +551,144 @@ def test_query_for_user_without_confirmed_mentor_program_families(self):
                              [])
 
-    def _assert_response_equals_json(self, query, expected_json):
-        with self.login(email=self.basic_user().email):
+    def test_judge_cannot_view_entrepreneur_mentor_profile(self):
+        judge = self._expert_user(UserRole.JUDGE)
+        with self.login(email=judge.email):
+            user = EntrepreneurFactory()
+            UserRoleContext(UserRole.MENTOR, user=user)
+            query = """
+                query{{
+                entrepreneurProfile(id:{id}) {{
+                    programRoles
+                }}
+            }}
+            """.format(id=user.id)
+
+            self._assert_error_in_response(query, NOT_ALLOWED_ACCESS_MESSAGE)
+
+    def test_user_without_role_can_view_staff_entrepreneur_profile(self):
+        current_user = self._expert_user()
+        user = EntrepreneurFactory()
+        UserRoleContext(UserRole.STAFF, user=user)
+        query = """
+            query{{
+                entrepreneurProfile(id:{id}) {{
+                    user{{lastName}}
+                }}
+            }}
+        """.format(id=user.id)
+        expected_json = {
+            'data': {
+                'entrepreneurProfile': {
+                    'user': {
+                        'lastName': user.last_name
+                    }
+                }
+            }
+        }
+        self._assert_response_equals_json(
+            query, expected_json, email=current_user.email)
+
+    def test_allowed_user_can_view_mentor_entrepreneur_profile(self):
+        allowed_user_roles = [
+            UserRole.FINALIST,
+            UserRole.AIR,
+            UserRole.MENTOR,
+            UserRole.PARTNER,
+            UserRole.ALUM
+        ]
+        for role in allowed_user_roles:
+            current_user = self._expert_user(role)
+            user = EntrepreneurFactory()
+            UserRoleContext(UserRole.MENTOR, user=user)
+            query = """
+                query{{
+                    entrepreneurProfile(id:{id}) {{
+                        user{{lastName}}
+                    }}
+                }}
+            """.format(id=user.id)
+            expected_json = {
+                'data': {
+                    'entrepreneurProfile': {
+                        'user': {
+                            'lastName': user.last_name
+                        }
+                    }
+                }
+            }
+            self._assert_response_equals_json(
+                query, expected_json, email=current_user.email)
+
+    def test_user_cannot_view_profile_with_non_current_allowed_roles(self):
+        allowed_user = self._expert_user(UserRole.FINALIST)
+        with self.login(email=allowed_user.email):
+            user = EntrepreneurFactory()
+            UserRoleContext(
+                UserRole.MENTOR,
+                user=user,
+                program=ProgramFactory(program_status=ENDED_PROGRAM_STATUS))
+            query = """
```

Comment:
It looks like this repeated definition indicates that we have a common pattern which should have been extracted. Hopefully this will be caught and cleaned up the next time we touch this test case, not going to block on this today.